### PR TITLE
First draft of chain signature wallet api docs

### PR DESCRIPTION
**Pull request by DavidM-D**
[Rendered](https://github.com/near/near-fastauth-wallet/blob/dmd/chain_sig_docs/docs/chain_signature_api.org)

### CODE REVIEW

**Comment by encody** [comment]
Would the recommended key path for someone who wants to have the same address on Ethereum, BSC, and Polygon look like this then?

```
david.near,,,
```

Or would it be recommended to use the `ethereum` chain-tagged key for all of these?

File: docs/chain_signature_api.org [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+[Accept] [Reject]
+#+end_src
+
+The user must make an informed decision about whether this is an action they would like to take.
+
+#+begin_src
+Signed ✅
+#+end_src
+
+** Using an untyped domain key
```
Comment (volovyks) [inline docs/chain_signature_api.org:120]: Why do we want to have untyped keys?

**Review by volovyks — COMMENTED** [review]
All this makes sense. I have two concerns:
- let's invest time in exploring existing HD Path derivation standards: https://trezor.io/learn/a/what-is-bip44 Maybe existing standards will not work for us, but can be a source of ideas and optimizations.
- Is it guaranteed that a third party will not be able to prove that 2 accounts (`sh.near,bitcoin` and `sh.near,bitcoin,x.org`) belong to one user? I assume that this is possible since it's possible to derive public keys. We may need to add additional seeds to prevent this.

**Comment by DavidM-D** [comment]
Good point on the BIP44 thing. I'll take a look at that and at least write a section explaining why we're not doing it.

File: docs/chain_signature_api.org (hunk 1 above)
Comment (DavidM-D) [reply docs/chain_signature_api.org:120]: They're quite useful for signing things that aren't transactions. For example if you want to send a message to another chain, sign an asset or encrypt something. That being said, that could just be new types of keys.

**Comment by DavidM-D** [comment]
This has been updated to use canonical JSON and is now ready for review

**Comment by esaminu** [comment]
LGTM integrating in https://github.com/near/fast-auth-signer/pull/201

**Review by esaminu — APPROVED** [review]
LGTM, integrating in https://github.com/near/fast-auth-signer/pull/201

**Pull request merged**